shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -35,6 +39,21 @@ static FlutterLocale FlutterLocaleFromNSLocale(NSLocale* locale) {
   return flutterLocale;
 }
 
+// Returns the unresolved path of the executable for the current process.
+static std::optional<std::string> GetExecutablePath() {
+  // Determine the size of the buffer required.
+  uint32_t pathSize = 0;
+  if (_NSGetExecutablePath(nullptr, &pathSize) == 0) {
```

Comment:
Do we actually want the full path, rather than the basename?

Reply:
See above comment; the full path.